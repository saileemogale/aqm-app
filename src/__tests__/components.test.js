import React from 'react';
import { configure, shallow } from 'enzyme';
import AirQualityMonitoringComponent from '../components';
import Adapter from "enzyme-adapter-react-16";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { Server } from 'mock-socket';

configure({ adapter: new Adapter() });

const mockStore = configureMockStore();
const store = mockStore({});

describe("AirQualityMonitoringComponent", () => {
    it("should render my component", () => {
            const wrapper = shallow(
                <Provider store={store}>
                    <AirQualityMonitoringComponent />
                </Provider>
            )
            expect(wrapper.getElements()).toMatchSnapshot();
        
    });

    it("connect to websocket", () => {
        const fakeURL = 'ws://localhost:1234';
        const mockServer = new Server(fakeURL);

        mockServer.on('connection', socket => {
            socket.on('message', data => {
              t.is(data, 'test message from app', 'we have intercepted the message and can assert on it');
              socket.send('test message from mock server');
            });
        });
    });
});