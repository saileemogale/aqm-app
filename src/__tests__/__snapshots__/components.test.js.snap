// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirQualityMonitoringComponent should render my component 1`] = `
Array [
  <Context.Provider
    value={
      Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Subscription {
          "handleChangeWrapper": [Function],
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": null,
        },
      }
    }
  >
    <AirQualityMonitoringComponent />
  </Context.Provider>,
]
`;
